src/beliefpropagation/beliefpropagation.jl

```diff
@@ -169,48 +135,46 @@ function belief_propagation(
 end
 
 function belief_propagation(
-  tn::ITensorNetwork;
-  contract_kwargs=(; alg="density_matrix", output_structure=path_graph_structure, maxdim=1),
-  nvertices_per_partition=nothing,
-  npartitions=nothing,
-  subgraph_vertices=nothing,
-  niters=default_bp_niters(mts),
-  target_precision=nothing,
-  verbose=false,
+  ptn::PartitionedGraph;
+  itensor_constructor=inds_e -> ITensor[dense(delta(inds_e))],
+  kwargs...,
 )
-  mts = message_tensors(tn; nvertices_per_partition, npartitions, subgraph_vertices)
-  return belief_propagation(tn, mts; contract_kwargs, niters, target_precision, verbose)
+  mts = message_tensors(ptn; itensor_constructor)
+  return belief_propagation(ptn, mts; kwargs...)
 end
-
 """
 Given a subet of vertices of a given Tensor Network and the Message Tensors for that network, return a Dictionary with the involved subgraphs as keys and the vector of tensors associated with that subgraph as values
 Specifically, the contraction of the environment tensors and tn[vertices] will be a scalar.
 """
-function get_environment(tn::ITensorNetwork, mts::DataGraph, verts::Vector; dir=:in)
-  subgraphs = unique([find_subgraph(v, mts) for v in verts])
+function get_environment(ptn::PartitionedGraph, mts, verts::Vector; dir=:in)
```

Comment:
It seems like there should be two versions of this function, one which takes partition vertices and one that takes underlying graph vertices, which could just call the one that takes partition vertices.

Reply:
Also, could you explain the `dir` keyword argument?

Reply:
I agree that there should now be two versions of this function. Let me work that out. The `dir` argument is whether you want the environment tensors coming on to `verts` or whether you want the environment tensors from `verts` to the rest of the network. This could be dropped though as it is confusing and could be handled by just directly passing `setdiff(vertives(ptn), verts)` by the user.

Reply:
Yeah, best to drop that.